### PR TITLE
Geolocate workshop location at create/update time and store in the DB

### PR DESCRIPTION
**Pull request by aoby**
rather than in the map client so we don't exceed Google's limit.

### CODE REVIEW

File: dashboard/app/models/pd/workshop.rb [hunk 1]
```diff
@@ -2,20 +2,21 @@
...
+#  course             :string(255)      not null
+#  subject            :string(255)
+#  capacity           :integer          not null
+#  notes              :text(65535)
+#  section_id         :integer
+#  started_at         :datetime
+#  ended_at           :datetime
+#  created_at         :datetime
+#  updated_at         :datetime
+#  processed_location :text(65535)
```
Comment (ashercodeorg) [inline dashboard/app/models/pd/workshop.rb]: Can this be ordered after `location_address` to keep similar fields near each other?

Comment (aoby) [inline dashboard/app/models/pd/workshop.rb]: done.

File: dashboard/app/models/pd/workshop.rb [hunk 2]
```diff
@@ -2,20 +2,21 @@
...
+#  course             :string(255)      not null
+#  subject            :string(255)
+#  capacity           :integer          not null
+#  notes              :text(65535)
+#  section_id         :integer
+#  started_at         :datetime
+#  ended_at           :datetime
+#  created_at         :datetime
+#  updated_at         :datetime
```
Comment (aoby) [inline dashboard/app/models/pd/workshop.rb:19]: strange. It reordered in the schema (and DB), but not in the annotations. Fixed.

**Comment by tanyaparker** [comment]
LGTM

File: dashboard/db/migrate/20160614000000_add_processed_location_to_pd_workshops.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+class AddProcessedLocationToPdWorkshops < ActiveRecord::Migration
+  def up
+    add_column :pd_workshops, :processed_location, :text, after: :location_address
+
+    Pd::Workshop.reset_column_information
+    Pd::Workshop.find_each do |workshop|
+      begin
+        workshop.processed_location = Pd::Workshop.process_location(workshop.location_address)
+        workshop.save!
+      rescue RuntimeError => e
```
Comment (aoby) [inline dashboard/db/migrate/20160614000000_add_processed_location_to_pd_workshops.rb:10]: Note: this should have been `StandardError`, not `RuntimeError`, to catch all the exceptions. See http://ruby-doc.org/core-2.1.1/Exception.html

@breville encountered a local error due to an invalid workshop that blew up the migration. I just made sure data in prod is valid and ready for the migration.

**Pull request merged**